dask_cuda/utils.py

```diff
@@ -357,3 +362,38 @@ def wait_workers(
             return False
         else:
             time.sleep(0.1)
+
+
+async def _all_to_all(client, cleanup=True):
+    """
+    Trigger all to all communication between workers and scheduler
+    """
+    workers = list(client.scheduler_info()["workers"])
+    futs = []
+    for w in workers:
+        bit_of_data = b"0" * 1
+        data = client.map(lambda x: bit_of_data, range(1), pure=False, workers=[w])
+        futs.append(data[0])
+
+    await wait(futs)
+
+    def f(x, y):
+        pass
+
+    new_futs = []
+    for w in workers:
+        for fut1, fut2 in combinations(futs, 2):
+            data = client.submit(f, fut1, fut2, workers=[w], pure=False)
+            new_futs.append(data)
+
+    await wait(new_futs)
+
+    if cleanup:
+        await client.cancel(futs)
+        await client.cancel(new_futs)
```

Comment:
I don't think I understand this, what's it useful for and when would you use `cleanup=False`?

Reply:
verifying in a test -- I can take it out if you don't want it

Reply:
I would be in favor of a cleaner function, which in a few months will just be some mysterious code that will be difficult to remember. However, if that's important for the test, it would be great to at least document what it does.